--- /usr/lib/python3.4/site-packages/nose/plugins/xunit.py	2015-03-12 19:27:34.122585895 +0000
+++ /usr/lib/python3.4/site-packages/nose/plugins/xunit.py	2015-03-12 19:28:02.309586714 +0000
@@ -65,6 +65,8 @@
     return xml_safe(cdata).replace(']]>', ']]>]]&gt;<![CDATA[')
 
 def id_split(idval):
+    idval = re.sub(r'array\(.*?\)', 'ndarray', re.sub('\n','', idval))
+    idval = re.sub(r'YTArray\(.*?\)', 'YTArray', idval)
     m = TEST_ID.match(idval)
     if m:
         name, fargs = m.groups()
--- /usr/lib/python2.7/site-packages/nose/plugins/xunit.py	2015-03-12 19:28:13.768587047 +0000
+++ /usr/lib/python2.7/site-packages/nose/plugins/xunit.py	2015-03-12 19:28:35.104587668 +0000
@@ -65,6 +65,8 @@
     return xml_safe(cdata).replace(']]>', ']]>]]&gt;<![CDATA[')
 
 def id_split(idval):
+    idval = re.sub(r'array\(.*?\)', 'ndarray', re.sub('\n','', idval))
+    idval = re.sub(r'YTArray\(.*?\)', 'YTArray', idval)
     m = TEST_ID.match(idval)
     if m:
         name, fargs = m.groups()
